jdaviz/app.py

```diff
@@ -109,47 +109,40 @@ def to_unit(self, data, cid, values, original_units, target_units):
             except RuntimeError:
                 eqv = []
             else:
+
+                # If there are only two values, this is likely the limits being converted, so then
+                # in case we need to use the spectral density equivalency, we need to provide only
+                # to spectral axis values. If there is only one value
+                if not np.isscalar(values) and len(values) == 2:
```

Comment:
Is this safer?
```suggestion
                if not np.isscalar(values) and len(values) > 1:
```

Reply:
No it's only for length 2 arrays, which is when we convert min/max ranges

Reply:
It's a little more complicated - in fact the code as written, without this clause, will only work properly if an array with the same length as the spectral axis is passed in. This clause makes it possible to also deal with the case where min/max values are passed in in which case we use the first spectral axis value in the conversion. But if we were to pass a length 5 array and the spectra axis had length 10 it's not clear what should happen.

In any case I think this discussion belongs in a separate issue/PR as it's beyond the scope of the simple fix here which preserves the original length 2 check.